src/background/service-worker-globals.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Some third-party libraries expect `window` (and sometimes `self`) to be defined.
+ * Service workers do not provide a window object, so expose the global scope to keep them happy.
+ */
+if (typeof globalThis.window === 'undefined') {
+  globalThis.window = globalThis
+}
```

Comment:
you should not use window in service-worker, because it could not have access to this global.
That's why I refactored alpheios-core to 3.5